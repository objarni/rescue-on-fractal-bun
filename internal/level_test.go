package internal

import (
	"fmt"
	approvals "github.com/approvals/go-approval-tests"
	"github.com/faiface/pixel"
	"objarni/rescue-on-fractal-bun/tests"
	"sort"
	"strings"
	"testing"
)

func init() {
	approvals.UseReporter(tests.ReportWithMeld())
}

func TestMapSignBuilder(t *testing.T) {
	var levels = map[string]Level{
		"Hembyn": {
			SignPosts: []SignPost{
				{
					Pos:  pixel.Vec{100, 10},
					Text: "Hembyn",
				},
				{
					Pos:  pixel.Vec{1000, 10},
					Text: "Skogen",
				},
			},
		},
		"Korsningen": {
			SignPosts: []SignPost{
				{
					Pos:  pixel.Vec{100, 10},
					Text: "Hembyn",
				},
			},
		},
	}

	approvals.VerifyString(t, mapSignsToString(BuildMapSignArray(levels)))
}

func mapSignsToString(signs []MapSign) string {
	descriptions := make([]string, 0)
	printVec := func(vec pixel.Vec) string {
		return fmt.Sprintf("<%v, %v>", vec.X, vec.Y)
	}
	for ix, sign := range signs {
		mapSignDescription := ""
		mapSignDescription += fmt.Sprintf("MapSign %v:", ix+1)
		mapSignDescription += fmt.Sprintf(" Position on map %v", printVec(sign.MapPos))
		mapSignDescription += fmt.Sprintf(" Links to %v %v\n", sign.LevelName, printVec(sign.LevelPos))
		descriptions = append(descriptions, mapSignDescription)
	}
	fmt.Print(descriptions)
	var sorted []string = descriptions
	sort.Strings(sorted)
	fmt.Print(sorted)
	return strings.Join(sorted, "\n")
}
