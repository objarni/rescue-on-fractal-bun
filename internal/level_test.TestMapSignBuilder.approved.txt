MapSign 1: Position on map <246, 109> Links to Hembyn <100, 10>

MapSign 2: Position on map <299, 375> Links to Hembyn <1000, 10>

MapSign 3: Position on map <246, 109> Links to Korsningen <100, 10>
